packages/marketplace/src/components/ui/submit-app-wizard/submit-app-wizard.tsx

```diff
@@ -0,0 +1,107 @@
+import { StepSubmitAppSuccess } from './steps/step-submit-app-success'
+import { Loader, ModalHeader, Formik, FormikHelpers, ModalProps } from '@reapit/elements'
+import { Form } from '@reapit/elements'
+import { WizardStep, WizardStepComponent, SetWizardStep } from './types'
+import { formFields } from './form-fields'
+import { validationSchemas } from './validation-schema'
+
+const componentMap: Record<WizardStep, WizardStepComponent> = {
+  BEFORE_YOU_START: StepBeforeYouStart,
+  INPUT_APP_NAME: StepInputAppName,
+  CREATE_NEW_APP: StepCreateNewApp,
+  INPUT_AUTHENTICATION_URIS: StepInputAuthenticationUris,
+  GRANT_PERMISSION: StepGrantPermissions,
+  SUBMIT_APP_SUCCESS: StepSubmitAppSuccess,
+}
+
+const titleMap: Record<WizardStep, string> = {
+  BEFORE_YOU_START: 'Before you start',
+  INPUT_APP_NAME: 'App Name',
+  CREATE_NEW_APP: 'Create a new app',
+  INPUT_AUTHENTICATION_URIS: 'Authentication',
+  GRANT_PERMISSION: 'Permissions',
+  SUBMIT_APP_SUCCESS: 'Success',
+}
+
+const { nameField, redirectUrisField, signoutUrisField, scopesField } = formFields
+
+export type HandleSubmitParams = {
+  dispatch: Dispatch
+  setWizardStep: SetWizardStep
+  afterClose: (() => void) | undefined
+}
+
+export const handleSubmit = ({ dispatch, setWizardStep, afterClose }: HandleSubmitParams) => (
+  values: CustomCreateAppModel,
+  actions: FormikHelpers<CustomCreateAppModel>,
+) => {
+  const { redirectUris, signoutUris, ...otherData } = values
+  const appToSubmit = {
+    ...otherData,
+    redirectUris: redirectUris ? redirectUris.split(',') : [],
+    signoutUris: signoutUris ? signoutUris.split(',') : [],
+  }
+
+  dispatch(submitApp({ ...appToSubmit, setErrors: actions.setErrors, setWizardStep, afterClose }))
+}
+
+// next button on step component is a submit button
+// will trigger onSubmit to save the state into snapshot -> set value to initialValues of formik
+// doing that make the state persist after the <Field> component unmount
+// follow offical guide how to implement mutil steps: https://github.com/jaredpalmer/formik/blob/master/examples/MultistepWizard.js
+// rerender form on intializeValues change -> set state of current form = initialValues
+export const handleUpdateFormState = setFormState => values => {
+  setFormState(values)
+}
+
+const initialFormValues = {
+  [nameField.name]: '',
+  [redirectUrisField.name]: '',
+  [signoutUrisField.name]: '',
+  [scopesField.name]: [],
+}
+
+export const SubmitAppWizard: React.FC<Pick<ModalProps, 'afterClose'>> = ({ afterClose }) => {
+  const [currentWizardStep, setWizardStep] = useState<WizardStep>('INPUT_APP_NAME')
+  const dispatch = useDispatch()
+  const isSubmitAppLoading = useSelector(selectSubmitAppLoadingState)
+
+  const CurrentStepComponent = componentMap[currentWizardStep as WizardStep]
+
+  if (isSubmitAppLoading) {
+    return <Loader />
+  }
+
+  if (!CurrentStepComponent) {
+    return null
+  }
+
+  return (
+    <div>
+      <Formik
+        enableReinitialize={true}
```

Comment:
No need to use `enableReinitialize` anymore because we won't ever change the initialValues